tests/unit/lms/services/blackboard_api/_schemas_test.py

```diff
@@ -0,0 +1,85 @@
+import pytest
+
+from lms.services.blackboard_api._schemas import BlackboardListFilesSchema
+from lms.validation._exceptions import ValidationError
+from tests import factories
+
+
+class TestBlackboardListFilesSchema:
+    def test_valid(self, json_data):
+        schema = BlackboardListFilesSchema(
+            factories.requests.Response(json_data=json_data)
+        )
+
+        result = schema.parse()
+
+        assert result == [
+            {
+                "id": "_7851_0",
+                "updated_at": "2008-05-06T07:26:35.000z",
+                "display_name": "File_0.pdf",
+            },
+            {
+                "id": "_7851_1",
+                "updated_at": "1983-05-26T02:37:23.000z",
+                "display_name": "File_1.pdf",
+            },
+        ]
+
+    def test_it_raises_if_results_is_missing(self, json_data, assert_raises):
+        del json_data["results"]
+
+        assert_raises()
+
+    def test_it_raises_if_results_isnt_a_list(self, json_data, assert_raises):
+        json_data["results"] = None
+
+        assert_raises()
+
+    @pytest.mark.parametrize("missing_field", ["id", "modified", "name"])
+    def test_it_raises_if_a_required_field_is_missing(
```

Comment:
I wonder was the right pytest pattern to generalize this when we have X schemas to test. A base TestSchema class that implements a few of these?

Reply:
Dunno, but I suspect there's something like that possible. I can certainly see the value in figuring out a good way to unit test schemas and adding some kind of test helpers for it, as we can then re-use that for all our schema tests. But I dunno exactly what it'd be yet. Sharing test fixtures between tests in `tests/unit/lms/validation/` and tests in `tests/unit/lms/services/blackboard_api/` would unfortunately be made harder by the location of these schemas in the `services/blackboard_api/` dir instead of in the `validation/` dir. Any shared fixtures would have to go in the very top-level `tests/unit/lms/conftest.py` file. I think that might turn out to be a downside of the "localized" layout here